### PR TITLE
Docs: add Docker version/API version matrix

### PR DESCRIPTION
**Pull request by thaJeztah**
Add a table to the documentation to find which API-version relates to which Docker version

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @moxiegirl @SvenDowideit :-)

File: docs/reference/api/docker_remote_api.md [hunk 1]
```diff
@@ -27,6 +27,25 @@ The current version of the API is v1.21 which means calling `/info` is the same
...
+1.7.x           | [1.19](/reference/api/docker_remote_api_v1.19/) | [API changes](/reference/api/docker_remote_api/#v1-19-api-changes)
+1.6.x           | [1.18](/reference/api/docker_remote_api_v1.18/) | [API changes](/reference/api/docker_remote_api/#v1-18-api-changes)
+1.5.x           | [1.17](/reference/api/docker_remote_api_v1.17/) | [API changes](/reference/api/docker_remote_api/#v1-17-api-changes)
+1.4.x           | [1.16](/reference/api/docker_remote_api_v1.16/) | [API changes](/reference/api/docker_remote_api/#v1-16-api-changes)
+1.3.x           | [1.15](/reference/api/docker_remote_api_v1.15/) | [API changes](/reference/api/docker_remote_api/#v1-15-api-changes)
+1.2.x           | [1.14](/reference/api/docker_remote_api_v1.14/) | [API changes](/reference/api/docker_remote_api/#v1-14-api-changes)
+1.1.x           | [1.13](/reference/api/docker_remote_api_v1.13/) | [API changes](https://github.com/docker/docker/blob/v1.1.2/docs/sources/reference/api/docker_remote_api.md#whats-new)
+1.0.x           | [1.12](/reference/api/docker_remote_api_v1.12/) | [API changes](https://github.com/docker/docker/blob/v1.0.1/docs/sources/reference/api/docker_remote_api.md#whats-new)
```
Comment (phemmer) [inline docs/reference/api/docker_remote_api.md]: Having a column which says the same thing for every row feels a little funny. How about something like:

Docker version  | API version                                     
----------------|-------------------------------------------------
1.9.x           | [1.21](/reference/api/docker_remote_api_v1.21/)  <sup>([changes](/reference/api/docker_remote_api/#v1-21-api-changes))</sup>
1.8.x           | [1.20](/reference/api/docker_remote_api_v1.20/) <sup>([changes](/reference/api/docker_remote_api/#v1-20-api-changes))</sup>

**Comment by phemmer** [comment]
A little unrelated, but since it seems the docker minor version, and the API version are a one-to-one mapping, why are they not the same? Why does docker 1.9.x not use API 1.9?

File: docs/reference/api/docker_remote_api.md (hunk 1 above)
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api.md]: yeah, was actually thinking of doing that first, the "source" was a bit cleaner to read when putting it in a separate column, so I opted for that.

I'll see what the others think, and will change if they agree :+1: 

Thanks for looking!

**Comment by thaJeztah** [comment]
> Why does docker 1.9.x not use API 1.9?

Beats me :D I always have to look up which API version relates to which Docker version (hence this table :))

File: docs/reference/api/docker_remote_api.md [hunk 2]
```diff
@@ -27,6 +27,25 @@ The current version of the API is v1.21 which means calling `/info` is the same
...
+1.8.x           | [1.20](/reference/api/docker_remote_api_v1.20/) | [API changes](/reference/api/docker_remote_api/#v1-20-api-changes)
+1.7.x           | [1.19](/reference/api/docker_remote_api_v1.19/) | [API changes](/reference/api/docker_remote_api/#v1-19-api-changes)
+1.6.x           | [1.18](/reference/api/docker_remote_api_v1.18/) | [API changes](/reference/api/docker_remote_api/#v1-18-api-changes)
+1.5.x           | [1.17](/reference/api/docker_remote_api_v1.17/) | [API changes](/reference/api/docker_remote_api/#v1-17-api-changes)
+1.4.x           | [1.16](/reference/api/docker_remote_api_v1.16/) | [API changes](/reference/api/docker_remote_api/#v1-16-api-changes)
+1.3.x           | [1.15](/reference/api/docker_remote_api_v1.15/) | [API changes](/reference/api/docker_remote_api/#v1-15-api-changes)
+1.2.x           | [1.14](/reference/api/docker_remote_api_v1.14/) | [API changes](/reference/api/docker_remote_api/#v1-14-api-changes)
+1.1.x           | [1.13](/reference/api/docker_remote_api_v1.13/) | [API changes](https://github.com/docker/docker/blob/v1.1.2/docs/sources/reference/api/docker_remote_api.md#whats-new)
```
Comment (moxiegirl) [inline docs/reference/api/docker_remote_api.md]: Nice. 1.13 and 1.12 aren't kept on the archive.  This should only go back to 1.14.

Comment (thaJeztah) [inline docs/reference/api/docker_remote_api.md]: Was in doubt there; wether or not I should go back to the first non-beta release. Will remove those

File: docs/reference/api/docker_remote_api.md (hunk 1 above)
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api.md]: @moxiegirl any preference on this one?

File: docs/reference/api/docker_remote_api.md (hunk 2 above)
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api.md]: removed the 1.13 and 1.12 rows

**Comment by cpuguy83** [comment]
@phemmer So that they don't have to be 1-to-1 if they don't need to be.

**Comment by phemmer** [comment]
> @phemmer So that they don't have to be 1-to-1 if they don't need to be.

I would still think the 2 can be related.
* API version would never be ahead of docker version, as changing the API requires changing docker itself.
* If the API doesn't change between docker versions, the API can keep the older version number. E.G. Docker 1.9.0 could use API 1.8.0.  
And would this really ever happen? Would there ever be a minor release that doesn't change the API?

**Comment by thaJeztah** [comment]
ping @moxiegirl ptal, also let me know if you like the suggestion @phemmer made https://github.com/docker/docker/pull/16764#discussion_r41149870, or if the current layout is okay to you

**Comment by moxiegirl** [comment]
@thaJeztah I think that is a case of six of one and half dozen of the other. We now have footnotes that all say the same thing.  UX wise --- the columns are easier to hit for more people. So, I say go with the column.

**Comment by moxiegirl** [comment]
LGTM as is.

**Comment by thaJeztah** [comment]
Author LGTM, merging as-is, but thanks for the suggestion @phemmer, we can always change in the future

**Pull request merged**